rewrite-java/src/main/java/org/openrewrite/java/UsePrecompiledRegExpForReplaceAll.java

```diff
@@ -0,0 +1,119 @@
+
+        @Override
+        public J.Block visitBlock(final J.Block block, final ExecutionContext executionContext) {
+            final J.Block processedBlock = super.visitBlock(block, executionContext);
+            final Cursor parentOrThrow = getCursor().getParentOrThrow();
+            if (parentOrThrow.getValue() instanceof J.ClassDeclaration) {
+                final Object regExp = parentOrThrow.pollNearestMessage(REG_EXP_KEY);
+                if (regExp == null) {
+                    return processedBlock;
+                }
+                patternProcessingInProgress.set(false);
+
+                final String regExpName = parentOrThrow.pollNearestMessage(REG_EXP_KEY + "_NAME");
+                return super.visitBlock(
+                        JavaTemplate
+                                .builder("private static final java.util.regex.Pattern " + regExpName + " = Pattern.compile(#{});")
+                                .contextSensitive()
+                                .imports("java.util.regex.Pattern")
+                                .build()
+                                .apply(
+                                        getCursor(),
+                                        processedBlock.getCoordinates().firstStatement(),
+                                        regExp),
+                        executionContext);
+            }
+            return processedBlock;
+        }
+
+        @Override
+        public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext executionContext) {
+            final boolean alreadyProcessingAPattern = patternProcessingInProgress.get();
+            if (!REPLACE_ALL_METHOD_MATCHER.matches(method) || alreadyProcessingAPattern) {
+                return super.visitMethodInvocation(method, executionContext);
+            }
+
+            patternProcessingInProgress.set(true);
+
+            final J.MethodInvocation invocation = super.visitMethodInvocation(method, executionContext);
+            final String regexPatternVariableName = VariableNameUtils.generateVariableName(PATTERN_VAR, getCursor(), VariableNameUtils.GenerationStrategy.INCREMENT_NUMBER);
+            final Object regexPatternIdentifier = new J.Identifier(
+                    randomId(),
+                    Space.SINGLE_SPACE,
+                    Markers.EMPTY,
+                    regexPatternVariableName,
+                    JavaType.buildType(Pattern.class.getName()),
+                    null);
+
+            final Cursor cursor = getCursor();
+            cursor.putMessageOnFirstEnclosing(J.ClassDeclaration.class, REG_EXP_KEY, invocation.getArguments().get(0));
+            cursor.putMessageOnFirstEnclosing(J.ClassDeclaration.class, REG_EXP_KEY + "_NAME", regexPatternVariableName);
+
+            return super.visitMethodInvocation(MATCHER_TEMPLATE.apply(
```

Comment:
When you make the change above, you may no longer need the `super.visitMethodInvocation()` here anymore. Or possibly you don't need the call on 98?